### ISSUE TITLE
Windows Builds: Appveyor integration

### ISSUE THREAD

**Issue by karissa** [deltachat/deltachat-desktop#25]
This will create windows builds. Depends upon a windows build script for the deltachat-core and deltachat-node libraries. We should use Appveyor for testing and building the executable.

**Comment by ralphtheninja** [comment]
Postponing this since there are multiple steps below that needs fixing before we can even address it.

**Comment by ralphtheninja** [comment]
Moving this down a notch. It's not realistic until we have a working core on windows.

**Comment by ralphtheninja** [comment]
Superseded by https://github.com/deltachat/deltachat-desktop/issues/606

**Comment by ralphtheninja** [comment]
Actually, re-opening because it's really something related to CI.

**Comment by homdx** [comment]
Builded binary deltachat-core need testing:
[deltachat/deltachat-core#306 (comment)](https://github.com/deltachat/deltachat-core/issues/306#issuecomment-461026117)
Possible problems: Windows does not support utf-8 coding, windows uses ansi encoding. (need fix).

**Comment by homdx** [comment]
@ralphtheninja I work with CI. But need fix to build version 0.40 and I will fix and continue work with Travis for full done compile)))
https://github.com/deltachat/deltachat-desktop/issues/606#issuecomment-465064978

**Comment by ralphtheninja** [comment]
@Jikstra I don't think we need this to build for windows since `electron-builder` can fix this anyway.

**Comment by Jikstra** [comment]
@ralphtheninja how do you mean? Can electron-builder download the windows prebuilts and just include them?

**Comment by ralphtheninja** [comment]
@Jikstra `npm install` is enough to download the prebuilts

**Comment by homdx** [comment]
> @Jikstra `npm install` is enough to download the prebuilts

npm install ./package.tgz

```
install a package that is sitting on the filesystem. Note: if you just want to link a dev directory into your npm root, you can do this more easily by using npm link.

Tarball requirements:

    The filename must use .tar, .tar.gz, or .tgz as the extension.
    The package contents should reside in a subfolder inside the tarball (usually it is called package/). npm strips one directory layer when installing the package (an equivalent of tar x --strip-components=1 is run).
    The package must contain a package.json file with name and version properties.
```

I think, are need to pack prebuild windows module to file tgz and install with download from release page on this github

**Comment by ralphtheninja** [comment]
All prebuilt binaries in the node bindings have been published to npm so a simple `npm install` and `npm run dev` should be enough to start desktop on linux, mac and windows.

It should be enough to do `electron-builder build -w` to package it for windows, and this can be done on Travis, so I don't think we need to use Appveyor at all.

**Comment by homdx** [comment]
@ralphtheninja @Jikstra Tested compiled version for Windows x64. Cant compile to one file with `npm run pack` this is give are error.
But npm run dev running under Windows is compiled and running. Please test this version
https://github.com/homdx/deltachat-windows/releases/tag/0.0.104

**Comment by homdx** [comment]
Completed work on Travis CI for Windows build:
https://travis-ci.org/homdx/deltachat-windows

![Travis result](https://user-images.githubusercontent.com/37062532/59927496-43a43500-9445-11e9-9842-70937a1c8db5.png)


Travis for windows:
https://github.com/homdx/deltachat-windows/blob/master/.travis.yml

**Comment by ralphtheninja** [comment]
Hi, that's cool. But _building_ the node bindings isn't necessary for desktop on windows since they have already been built. I've said this multiple times already :wink:

**Comment by homdx** [comment]
**_HelIo!_**
In my tested repository (after almost of 200 commits), I release:

1. First repeatable build for DeltaChat for windows (used cached version rust, sources, after auto upload from travis-ci to repository, you can repeat build old version:)
2. Also, you can run and full rebuild deltachat-desktop for windows from latest sources

@ralphtheninja I don't know, why `npm run pack` not working. You can sea travis logs in my repository)
@Jikstra If you trust my build and can repeat it. You can add version 0.0.104 for windows binary release. Or version for windows are will release in next release?

Windows users and Developers can download and test:
https://github.com/homdx/test-win-build/releases

**Comment by Jikstra** [comment]
hey @homdx,

thanks for your work, your repository seems a bit confusing to me, it still has all the old .cmd/.sh files for the c core in it? I think the only relevant parts are the .travis.yml and make-build-cache.sh ? How and why can you run a .sh on a windows travis? Don't know much about windows, currently i'm trying to compile deltachat-core-rust in a windows 10 virtual machine. I think this whole process should get a lot simpler as soon as we have win64 prebuilts. But i think it still makes sense to have all the build tools on the ci installed as deltachat-node falls back to from source building if there's no prebuilt.

**Comment by homdx** [comment]
@Jikstra I refactoring code, please see (remove old builder for C node deltachat)
And pushed and released new 0.105 version
https://github.com/homdx/test-win-build/releases
